tests/v2/validation/rbac/configmaps/configmaps_rbac_test.go

```diff
@@ -0,0 +1,276 @@
+				require.NoError(cm.T(), err)
+				_, err = dep.CreateDeployment(standardUserClient, cm.cluster.ID, deploymentName, namespace.Name, newPodTemplate(configMapCreatedByUser.Name, ""), 1)
+
+			case rbac.ClusterMember.String(), rbac.ReadOnly.String():
+				require.Error(cm.T(), err)
+				require.True(cm.T(), k8sError.IsForbidden(err))
+			}
+		})
+	}
+}
+
+func (cm *ConfigmapsRBACTestSuite) TestListConfigmaps() {
+	subSession := cm.session.NewSession()
+	defer subSession.Cleanup()
+
+	tests := []struct {
+		role   rbac.Role
+		member string
+	}{
+		{rbac.ClusterOwner, rbac.StandardUser.String()},
+		{rbac.ClusterMember, rbac.StandardUser.String()},
+		{rbac.ProjectOwner, rbac.StandardUser.String()},
+		{rbac.ProjectMember, rbac.StandardUser.String()},
+		{rbac.ReadOnly, rbac.StandardUser.String()},
+	}
+	for _, tt := range tests {
+		cm.Run("Validate config map creation for user with role "+tt.role.String(), func() {
+			adminProject, namespace, err := createProjectAndNamespace(cm.client, cm.cluster.ID)
+			require.NoError(cm.T(), err)
+
+			newUser, standardUserClient, err := rbac.CreateUserAndAddToClusterRole(cm.client, tt.member, tt.role.String(), cm.cluster, adminProject)
+			require.NoError(cm.T(), err)
+			cm.T().Logf("Created user: %v", newUser.Username)
+
+			adminWranglerContext, err := cm.client.WranglerContext.DownStreamClusterWranglerContext(cm.cluster.ID)
+			require.NoError(cm.T(), err)
+			configMapCreatedByAdmin, err := createConfigmap(namespace, adminWranglerContext)
+
+			downstreamWranglerContext, err := standardUserClient.WranglerContext.DownStreamClusterWranglerContext(cm.cluster.ID)
+			configMapListAsUser, err := downstreamWranglerContext.Core.ConfigMap().List(namespace.Name, metav1.ListOptions{
+				FieldSelector: "metadata.name=" + configMapCreatedByAdmin.Name,
+			})
+
+			switch tt.role.String() {
+			case rbac.ClusterOwner.String(), rbac.ProjectOwner.String(), rbac.ProjectMember.String():
+				require.NoError(cm.T(), err)
+				_, err = dep.CreateDeployment(standardUserClient, cm.cluster.ID, deploymentName, namespace.Name, newPodTemplate(configMapCreatedByAdmin.Name, ""), 1)
+				require.NoError(cm.T(), err)
+				require.Equal(cm.T(), len(configMapListAsUser.Items), 1)
+			case rbac.ClusterMember.String(), rbac.ReadOnly.String():
+				require.Error(cm.T(), err)
+				require.True(cm.T(), k8sError.IsForbidden(err))
+			}
+		})
+	}
+}
+
+func (cm *ConfigmapsRBACTestSuite) TestDeleteConfigmap() {
+	subSession := cm.session.NewSession()
+	defer subSession.Cleanup()
+
+	tests := []struct {
+		role   rbac.Role
+		member string
+	}{
+		{rbac.ClusterOwner, rbac.StandardUser.String()},
+		{rbac.ClusterMember, rbac.StandardUser.String()},
+		{rbac.ProjectOwner, rbac.StandardUser.String()},
+		{rbac.ProjectMember, rbac.StandardUser.String()},
+		{rbac.ReadOnly, rbac.StandardUser.String()},
+	}
+
+	for _, tt := range tests {
+		cm.Run("Validate config map creation for user with role "+tt.role.String(), func() {
```

Comment:
Can you change this to "Validate config map can be deleted by user with role"

Reply:
Done